src/pip/_internal/resolution/resolvelib/factory.py

```diff
@@ -127,40 +130,45 @@ def iter_found_candidates(self, ireq, extras):
         # requirement needs to return only one candidate per version, so we
         # implement that logic here so that requirements using this helper
         # don't all have to do the same thing later.
-        seen_versions = set()  # type: Set[_BaseVersion]
+        version_candidates = collections.OrderedDict(
+        )  # type: Dict[_BaseVersion, Candidate]
```

Comment:
nit: The line breaking here is *horrible*. Is that really the only way to get within the line length? Maybe change the variable name to `results` (which is generic, but accurate - these are the results we're going to return, indexed by version). And maybe use a type alias, if that's not enough.

This is when I really start to hate inflexible linting rules...

Reply:
It is the only way I know :( But I agree 100% this is horrible, let me try “fix” this.

Reply:
Your fix works for me - thanks!